chain/src/chain.rs

```diff
@@ -117,7 +126,11 @@ impl<CI: ChainIndex + 'static> ChainService<CI> {
         }
     }
 
-    fn process_block(&mut self, block: Arc<Block>) -> Result<(), ProcessBlockError> {
+    fn process_block(
```

Comment:
This function intends to verify and insert a new block to store, I think that `cycles_set` is always passed as an empty vector, we should remove it from here and `insert_block` function below.